src/containers/CheckoutPage/CheckoutPageWithInquiryProcess.js

```diff
@@ -0,0 +1,285 @@
+import { Form as FinalForm } from 'react-final-form';
+import classNames from 'classnames';
+
+// Import contexts and util modules
+import { FormattedMessage, intlShape } from '../../util/reactIntl';
+import { displayPrice } from '../../util/configHelpers';
+import { pathByRouteName } from '../../util/routes';
+import { propTypes } from '../../util/types';
+import { formatMoney } from '../../util/currency';
+import { createSlug } from '../../util/urlHelpers';
+import { isTransactionInitiateListingNotFoundError } from '../../util/errors';
+import * as validators from '../../util/validators';
+import { getProcess } from '../../transactions/transaction';
+
+// Import shared components
+import {
+  FieldTextInput,
+  Form,
+  H3,
+  H4,
+  Heading,
+  NamedLink,
+  Page,
+  PrimaryButton,
+} from '../../components';
+
+import { getTransactionTypeData } from './CheckoutPageTransactionHelpers.js';
+
+import CustomTopbar from './CustomTopbar';
+import DetailsSideCard from './DetailsSideCard';
+import MobileListingImage from './MobileListingImage';
+
+import css from './CheckoutPage.module.css';
+
+const ErrorMessage = props => {
+  const { error } = props;
+
+  // Since the listing data is already given from the ListingPage
+  // and stored to handle refreshes, it might not have the possible
+  // deleted or closed information in it. If the transaction
+  // initiate or the speculative initiate fail due to the listing
+  // being deleted or closed, we should dig the information from the
+  // errors and not the listing data.
+  const listingNotFound = isTransactionInitiateListingNotFoundError(error);
+
+  // No transaction process attached to listing
+  const noTransactionProcessAlias = error?.message === 'No transaction process attached to listing';
+
+  return error ? (
+    <p className={css.error}>
+      {listingNotFound ? (
+        <FormattedMessage id="CheckoutPage.listingNotFoundError" />
+      ) : noTransactionProcessAlias ? (
+        <FormattedMessage id="CheckoutPageWithInquiryProcess.initiateInquiryErrorNoProcess" />
+      ) : (
+        <FormattedMessage id="CheckoutPageWithInquiryProcess.initiateInquiryError" />
+      )}
+    </p>
+  ) : null;
+};
+
+const handleSubmit = (submitting, setSubmitting, props) => values => {
+  if (submitting) {
+    return;
+  }
+  setSubmitting(true);
+
+  const {
+    history,
+    config,
+    routeConfiguration,
+    pageData,
+    processName,
+    onInquiryWithoutPayment,
+    onSubmitCallback,
+  } = props;
+
+  const { inquiryMessage } = values;
+
+  const { listingType, transactionProcessAlias, unitType } =
+    pageData?.listing?.attributes?.publicData || {};
+
+  const process = processName ? getProcess(processName) : null;
+  const transitions = process.transitions;
+  const transition = transitions.INQUIRE_WITHOUT_PAYMENT;
+
+  // These are the order parameters for the first payment-related transition
+  // which is either initiate-transition or initiate-transition-after-enquiry
```

Comment:
Comment refers to payment related transition, which should not be possible on CheckoutPageWithInquiryProcess